### ISSUE TITLE
Bug: Linter not working properly

### ISSUE THREAD

**Issue by peek4bUh** [platers/obsidian-linter#1186]
- [ x ] I have verified that I am on the latest version of the Linter

## Describe the Bug

Fresh install of the Linter trying to format a file like you did in the showcase but is not doing nothing, only formats the title and the yaml but not the content of the note. And `File on save` is not triggered but is enabled with the other option named `Display message on lint`.

## How to Reproduce

Example to reproduce issue with

``` markdown
---
id: '1/3'
language: english
parent: "[[computing-moc]]"
tags: [permanent]
title: Absolute Pathname
---

# absolute Pathname

An **absolute pathname** begins with the [[root-directory|root directory]] (`/`) and follows the tree branch by branch until the path to the desired directory or file completed.

## origin Asda Sda Asd Asd
asdasd asd asdka lskd ñal k k k                              a sda s as dasd ñalksdñ lkasñldk añslkd ñalksd ñlkaskdlñ asd as d asñka ñklsñk laklñs ñlaskdpwpaokwp k   kk apos podasdpok aposkd pokwpoa ko k               apkd pwap kapwk pakw pk kapk spkapk dspka
- asdasdadsasd
-    as      dasdas-
-           asd


For instance, `/home/tux/example.txt` tells you to start at the `/home` directory on your hard drive and then click into the `tux` directory, where you'll find a text file called `example.txt`.
```

Please make sure to include logs and the rule(s) that cause the issue if applicable. You can include logs by going to the debug tab of the Linter and setting the log level to debug. Then enable the capturing of logs and either lint on save or lint the current file. Then go back to the debug tab and you can grab Linter logs from the last run.

If you are not sure which rule(s) may cause the issue, please attach the `data.json` file located at `<vault>/.obsidian/plugins/obsidian-linter/` or copy the config from the debug tab.

[data.json](https://github.com/user-attachments/files/17297293/data.json)

## Expected Behavior

Format the file without blank spaces and with the `Ctrl + s` hotkey.

Expected output if applicable:

``` markdown
---
id: '1/3'
language: english
parent: "[[computing-moc]]"
tags: [permanent]
title: Absolute Pathname
---

# Absolute Pathname

An **absolute pathname** begins with the [[root-directory|root directory]] (`/`) and follows the tree branch by branch until the path to the desired directory or file completed.

## Origin Asda Sda Asd Asd

asdasd asd asdka lskd ñal k k k a sda s as dasd ñalksdñ lkasñldk añslkd ñalksd ñlkaskdlñ asd as d asñka ñklsñk laklñs ñlaskdpwpaokwp k   kk apos podasdpok aposkd pokwpoa ko k  apkd pwap kapwk pakw pk kapk spkapk dspka
- asdasdadsasd
- as dasdas-
- asd

For instance, `/home/tux/example.txt` tells you to start at the `/home` directory on your hard drive and then click into the `tux` directory, where you'll find a text file called `example.txt`.
```

## Screenshots
![image](https://github.com/user-attachments/assets/6ced8419-7e2b-4608-8237-0520293136c6)

## Device

- [ x ] Desktop
- [ ] Mobile

## Additional Context

None.

**Comment by peek4bUh** [comment]
Using the plugin named `Prettier Format` from Andrew Lisowski works

![image](https://github.com/user-attachments/assets/5c199d26-de60-4359-bff9-7ae8f75d9d94)

**Comment by pjkaufman** [comment]
Hey @peek4bUh , could you see if there are any errors in the console (Crtl+Shift+I and then console)? Also, what version of the Linter are you using?

**Comment by peek4bUh** [comment]
Hello @pjkaufman, sorry for late response, I'm using the v1.26.0 and here is an image of the console:

![image](https://github.com/user-attachments/assets/bb803839-0adb-4c3f-9a27-f7b9fc8c7a28)

**Comment by pjkaufman** [comment]
Thanks for doing that. No worries on the response time. It looks like there is not really an issue in the console which implies to me that the Linter did not actually run or the log level is so high that the logs I would expect to see to know what happened are missing. I don't even see the info about the Linter loading, but that could be caused by the log level being error or warn.

So a couple of things I would test are as follows (I will see about reproducing this when I get a chance, but this information will help as well):
- Does this issue happen if the Linter is the only enabled plugin (this will tell me if another plugin is also overriding the file save logic and thus interfering with the Linter)?
  - This will require either disabling and reenabling the Linter, reloading without saving, or closing and opening Obsidian again
- Can you set the log level under the Linter's settings' debug tab to debug/trace if it currently is not and then share the contents of the console after trying to run a lint?

**Comment by peek4bUh** [comment]
I did a new vault with only the Linter plugin and nothing, atm the `Ctrl + S` is now working and I didn't do anything, here is the log:
[obsidian.md-1728565084264.log](https://github.com/user-attachments/files/17328483/obsidian.md-1728565084264.log)

**Comment by lup3v0** [comment]
I have the same issue. I used to be able to lint on save (CMD + S) but it stopped working. When I assign to a hotkey, it works. So something is conflicting. Investigating.

**Comment by RedAtman** [comment]
same issue

**Comment by RedAtman** [comment]
same issue

**Comment by pjkaufman** [comment]
Hey @RedAtman , @peek4bUh , @lup3v0 , could I get a list of plugins you have installed? I can check to see if any of them are overriding the save action.

**Comment by lup3v0** [comment]
Thanks @pjkaufman.

Here is the list of plugins:

Archiver
Calendar
Callout Manager
Copy document as HTML
Custom Frames
Dataview
Day Planner
Excalidraw
Folder notes
Image Converter
Kanban
Linter
List Callouts
Meta Bind
Minimal Theme Settings
Natural Language Dates
Omnivore
Outliner
Paste URL into selection
Periodic Notes
Projects
Quick Switcher++
QuickAdd
Style Settings
Tasks
Templater
WordNet Dictionary

**Comment by pjkaufman** [comment]
@lup3v0 ,  TLDR: did not find the culprit, but if you check your hotkey list and use the hotkey finder for specific key combinations it should tell you which commands are assigned to `ctrl+s` (see image below).

It looks like Excalidraw does override the save function, but it looks like that does not prevent linting on save.
- https://github.com/zsviczian/obsidian-excalidraw-plugin/blob/4e2d7eb63799f55339b0a359ac317176a43da5a4/src/main.ts#L2853-L2863
- https://github.com/zsviczian/obsidian-excalidraw-plugin/blob/4e2d7eb63799f55339b0a359ac317176a43da5a4/src/main.ts#L1488-L1493

I think we can rule excalidraw out.

One thing to make sure of is that you see save the current file as a hotkey for `ctrl+s`:
![test](https://github.com/user-attachments/assets/b1cea86d-1a42-4b29-a570-96eb96836b6e)

If Save the current file is not there, it needs to be. It should also show the culprit.

I did look at the source code of most of the plugins listed above (I ruled out ones I use). So I recommend looking at your hotkeys and seeing which is conflicting with the save current file command.

**Comment by peek4bUh** [comment]
Sorry for late response, here is mine:

- Excalidraw
- Dataview
- Advanced Tables
- Style Settings
- Minimal Theme Settings
- Admonition
- Linter
- For note view mode
- Front Matter Title
- Prettier Format

**Comment by peek4bUh** [comment]
Now works, I don't understand anything :D

**Comment by pjkaufman** [comment]
Based on the information provided so far, I am planning to close this issue. Is there any lingering issue around this or has the issue gone away?

**Comment by lup3v0** [comment]
Thanks @pjkaufman. I thought I had responded earlier but apparently not. The issue has gone away on my end. I re-assigned the shortcuts, rebooted, and it is now working as intended. Thank you!

**Comment by pjkaufman** [comment]
Based on the information provided, it looks like the Linter is not the cause of the issue and it looks to be fixable by adding file save back to the `ctrl+s` shortcut. I am guessing it is `cmd+s` on Mac. So I am going to close out this issue.
